File: ios/Sources/AdvancedFileManagerPlugin/AdvancedFileManagerPlugin.swift
import Foundation
import Capacitor
import UIKit
import UniformTypeIdentifiers

/**
 * Advanced File Manager Plugin for Capacitor
 * 模块化架构的文件管理插件
 */
@objc(AdvancedFileManagerPlugin)
public class AdvancedFileManagerPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "AdvancedFileManagerPlugin"
    public let jsName = "AdvancedFileManager"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "requestPermissions", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "checkPermissions", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "listDirectory", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "createDirectory", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "deleteDirectory", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "createFile", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "readFile", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "writeFile", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "deleteFile", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "renameFile", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "moveFile", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "copyFile", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getFileInfo", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "exists", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "searchFiles", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "openSystemFilePicker", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "openSystemFileManager", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "openFileWithSystemApp", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    
    // 模块实例
    private let fileOps = FileOperations()
    private let dirOps = DirectoryOperations()
    private let fileSearcher = FileSearcher()

    // MARK: - 权限管理

    @objc public override func requestPermissions(_ call: CAPPluginCall) {
        call.resolve([
            "granted": true,
            "message": "File access permissions granted within app sandbox"
        ])
    }

    @objc public override func checkPermissions(_ call: CAPPluginCall) {
        call.resolve([
            "granted": true,
            "message": "File access permissions granted within app sandbox"
        ])
    }

    // MARK: - 目录操作

    @objc func listDirectory(_ call: CAPPluginCall) {
        guard let path = call.getString("path") else {
            call.reject("Path is required")
            return
        }

        let showHidden = call.getBool("showHidden") ?? false
        let sortBy = call.getString("sortBy") ?? "name"
        let sortOrder = call.getString("sortOrder") ?? "asc"

        do {
            let result = try dirOps.listDirectory(path: path, showHidden: showHidden, sortBy: sortBy, sortOrder: sortOrder)
            call.resolve(result)
        } catch {
            call.reject("Failed to list directory: \(error.localizedDescription)")
        }
    }

    @objc func createDirectory(_ call: CAPPluginCall) {
        guard let path = call.getString("path") else {
            call.reject("Path is required")
            return
        }

        let recursive = call.getBool("recursive") ?? false

        do {
            try dirOps.createDirectory(path: path, recursive: recursive)
            call.resolve()
        } catch {
            call.reject("Failed to create directory: \(error.localizedDescription)")
        }
    }

    @objc func deleteDirectory(_ call: CAPPluginCall) {
        guard let path = call.getString("path") else {
            call.reject("Path is required")
            return
        }

        do {
            try dirOps.deleteDirectory(path: path)
            call.resolve()
        } catch {
            call.reject("Failed to delete directory: \(error.localizedDescription)")
        }
    }

    // MARK: - 文件操作

    @objc func createFile(_ call: CAPPluginCall) {
        guard let path = call.getString("path") else {
            call.reject("Path is required")
            return
        }

        let content = call.getString("content") ?? ""
        let encoding = call.getString("encoding") ?? "utf8"

        do {
            try fileOps.createFile(path: path, content: content, encoding: encoding)
            call.resolve()
        } catch {
            call.reject("Failed to create file: \(error.localizedDescription)")
        }
    }

    @objc func readFile(_ call: CAPPluginCall) {
        guard let path = call.getString("path") else {
            call.reject("Path is required")
            return
        }

        let encoding = call.getString("encoding") ?? "utf8"

        do {
            let result = try fileOps.readFile(path: path, encoding: encoding)
            call.resolve(result)
        } catch {
            call.reject("Failed to read file: \(error.localizedDescription)")
        }
    }

    @objc func writeFile(_ call: CAPPluginCall) {
        guard let path = call.getString("path"),
              let content = call.getString("content") else {
            call.reject("Path and content are required")
            return
        }

        let encoding = call.getString("encoding") ?? "utf8"
        let append = call.getBool("append") ?? false

        do {
            try fileOps.writeFile(path: path, content: content, encoding: encoding, append: append)
            call.resolve()
        } catch {
            call.reject("Failed to write file: \(error.localizedDescription)")
        }
    }

    @objc func deleteFile(_ call: CAPPluginCall) {
        guard let path = call.getString("path") else {
            call.reject("Path is required")
            return
        }

        do {
            try fileOps.deleteFile(path: path)
            call.resolve()
        } catch {
            call.reject("Failed to delete file: \(error.localizedDescription)")
        }
    }

    @objc func renameFile(_ call: CAPPluginCall) {
        guard let path = call.getString("path"),
              let newName = call.getString("newName") else {
            call.reject("Path and new name are required")
            return
        }

        do {
            try fileOps.renameFile(path: path, newName: newName)
            call.resolve()
        } catch {
            call.reject("Failed to rename file: \(error.localizedDescription)")
        }
    }

    @objc func moveFile(_ call: CAPPluginCall) {
        guard let sourcePath = call.getString("sourcePath"),
              let destinationPath = call.getString("destinationPath") else {
            call.reject("sourcePath and destinationPath are required")
            return
        }

        do {
            try fileOps.moveFile(sourcePath: sourcePath, destinationPath: destinationPath)
            call.resolve()
        } catch {
            call.reject("Failed to move file: \(error.localizedDescription)")
        }
    }

    @objc func copyFile(_ call: CAPPluginCall) {
        guard let sourcePath = call.getString("sourcePath"),
              let destinationPath = call.getString("destinationPath") else {
            call.reject("sourcePath and destinationPath are required")
            return
        }

        let overwrite = call.getBool("overwrite") ?? false

        do {
            try fileOps.copyFile(sourcePath: sourcePath, destinationPath: destinationPath, overwrite: overwrite)
            call.resolve()
        } catch {
            call.reject("Failed to copy file: \(error.localizedDescription)")
        }
    }

    @objc func getFileInfo(_ call: CAPPluginCall) {
        guard let path = call.getString("path") else {
            call.reject("Path is required")
            return
        }

        do {
            let result = try fileOps.getFileInfo(path: path)
            call.resolve(result)
        } catch {
            call.reject("Failed to get file info: \(error.localizedDescription)")
        }
    }

    @objc func exists(_ call: CAPPluginCall) {
        guard let path = call.getString("path") else {
            call.reject("Path is required")
            return
        }

        call.resolve(["exists": fileOps.exists(path: path)])
    }

    // MARK: - 搜索

    @objc func searchFiles(_ call: CAPPluginCall) {
        guard let directory = call.getString("directory"),
              let query = call.getString("query") else {
            call.reject("Directory and query are required")
            return
        }

        let searchType = call.getString("searchType") ?? "name"
        let fileTypes = call.getArray("fileTypes")?.compactMap { $0 as? String }
        let maxResults = call.getInt("maxResults") ?? 100
        let recursive = call.getBool("recursive") ?? true

        do {
            let result = try fileSearcher.searchFiles(directory: directory, query: query,
                                                       searchType: searchType, fileTypes: fileTypes,
                                                       maxResults: maxResults, recursive: recursive)
            call.resolve(result)
        } catch {
            call.reject("Failed to search files: \(error.localizedDescription)")
        }
    }

    // MARK: - 系统文件选择器

    @objc func openSystemFilePicker(_ call: CAPPluginCall) {
        let type = call.getString("type") ?? "file"
        let multiple = call.getBool("multiple") ?? false

        DispatchQueue.main.async {
            if type == "directory" {
                self.openDirectoryPicker(call: call)
            } else {
                self.openFilePicker(call: call, multiple: multiple)
            }
        }
    }

    private func openFilePicker(call: CAPPluginCall, multiple: Bool) {
        let picker = UIDocumentPickerViewController(forOpeningContentTypes: [.item], asCopy: true)
        picker.allowsMultipleSelection = multiple
        picker.delegate = self
        
        bridge?.saveCall(call)
        bridge?.viewController?.present(picker, animated: true, completion: nil)
    }

    private func openDirectoryPicker(call: CAPPluginCall) {
        let picker = UIDocumentPickerViewController(forOpeningContentTypes: [.folder])
        picker.delegate = self
        
        bridge?.saveCall(call)
        bridge?.viewController?.present(picker, animated: true, completion: nil)
    }

    @objc func openSystemFileManager(_ call: CAPPluginCall) {
        // iOS 没有直接打开文件管理器的方式
        call.resolve()
    }

    @objc func openFileWithSystemApp(_ call: CAPPluginCall) {
        guard let filePath = call.getString("path") else {
            call.reject("File path is required")
            return
        }

        let url = URL(fileURLWithPath: filePath)

        guard FileManager.default.fileExists(atPath: filePath) else {
            call.reject("File does not exist: \(filePath)")
            return
        }

        DispatchQueue.main.async {
            let documentController = UIDocumentInteractionController(url: url)
            documentController.delegate = self
            
            if let viewController = self.bridge?.viewController {
                if !documentController.presentPreview(animated: true) {
                    documentController.presentOpenInMenu(from: viewController.view.bounds, in: viewController.view, animated: true)
                }
            }
            
            call.resolve()
        }
    }

    // MARK: - 工具方法

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve(["value": value])
    }
}

// MARK: - UIDocumentPickerDelegate

extension AdvancedFileManagerPlugin: UIDocumentPickerDelegate {
    public func documentPicker(_ controller: UIDocumentPickerViewController, didPickDocumentsAt urls: [URL]) {
        guard let call = bridge?.savedCall(withID: bridge?.savedCalls?.first?.key ?? "") else {
            return
        }
        
        var files: [[String: Any]] = []
        var directories: [[String: Any]] = []
        
        for url in urls {
            var isDirectory: ObjCBool = false
            FileManager.default.fileExists(atPath: url.path, isDirectory: &isDirectory)
            
            do {
                let resourceValues = try url.resourceValues(forKeys: [
                    .fileSizeKey,
                    .contentModificationDateKey,
                    .creationDateKey,
                    .isDirectoryKey
                ])
                
                let fileInfo = FileUtils.createFileInfo(from: url, resourceValues: resourceValues)
                
                if isDirectory.boolValue {
                    directories.append(fileInfo)
                } else {
                    files.append(fileInfo)
                }
            } catch {
                // 基本信息
                let info: [String: Any] = [
                    "name": url.lastPathComponent,
                    "path": url.path,
                    "type": isDirectory.boolValue ? "directory" : "file"
                ]
                
                if isDirectory.boolValue {
                    directories.append(info)
                } else {
                    files.append(info)
                }
            }
        }
        
        call.resolve([
            "files": files,
            "directories": directories,
            "cancelled": false
        ])
        
        bridge?.releaseCall(call)
    }
    
    public func documentPickerWasCancelled(_ controller: UIDocumentPickerViewController) {
        guard let call = bridge?.savedCall(withID: bridge?.savedCalls?.first?.key ?? "") else {
            return
        }
        
        call.resolve([
            "files": [],
            "directories": [],
            "cancelled": true
        ])
        
        bridge?.releaseCall(call)
    }
}

// MARK: - UIDocumentInteractionControllerDelegate

extension AdvancedFileManagerPlugin: UIDocumentInteractionControllerDelegate {
    public func documentInteractionControllerViewControllerForPreview(_ controller: UIDocumentInteractionController) -> UIViewController {
        return bridge?.viewController ?? UIViewController()
    }
}

File: android/src/main/java/com/mycompany/plugins/example/AdvancedFileManagerPlugin.java
package com.mycompany.plugins.example;

import android.content.Intent;
import android.net.Uri;
import android.os.Build;
import android.os.Environment;
import android.provider.DocumentsContract;
import android.webkit.MimeTypeMap;

import androidx.activity.result.ActivityResult;
import androidx.core.content.FileProvider;

import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.ActivityCallback;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.getcapacitor.annotation.Permission;
import com.getcapacitor.annotation.PermissionCallback;

import com.mycompany.plugins.example.core.FileOperations;
import com.mycompany.plugins.example.core.DirectoryOperations;
import com.mycompany.plugins.example.core.FileUtils;
import com.mycompany.plugins.example.permission.PermissionManager;
import com.mycompany.plugins.example.picker.SystemFilePicker;
import com.mycompany.plugins.example.search.FileSearcher;
import com.mycompany.plugins.example.ai.AIEditOperations;

import java.io.File;

/**
 * Advanced File Manager Plugin
 * 模块化架构的文件管理插件
 */
@CapacitorPlugin(
    name = "AdvancedFileManager",
    permissions = {
        @Permission(
            strings = { 
                android.Manifest.permission.READ_EXTERNAL_STORAGE, 
                android.Manifest.permission.WRITE_EXTERNAL_STORAGE 
            },
            alias = "storage"
        )
    }
)
public class AdvancedFileManagerPlugin extends Plugin {

    // 模块实例
    private FileOperations fileOps;
    private DirectoryOperations dirOps;
    private PermissionManager permManager;
    private SystemFilePicker filePicker;
    private FileSearcher fileSearcher;
    private AIEditOperations aiEditOps;

    @Override
    public void load() {
        super.load();
        // 初始化各模块
        fileOps = new FileOperations(getContext());
        dirOps = new DirectoryOperations(getContext());
        permManager = new PermissionManager(this);
        filePicker = new SystemFilePicker(this);
        fileSearcher = new FileSearcher(getContext());
        aiEditOps = new AIEditOperations(getContext());
    }

    // ==================== 权限管理 ====================

    @PluginMethod
    public void requestPermissions(PluginCall call) {
        if (permManager.hasStoragePermissions()) {
            call.resolve(permManager.createPermissionResult(true, "Storage permissions already granted"));
        } else {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
                permManager.requestManageExternalStoragePermission(call);
            } else {
                requestPermissionForAlias("storage", call, "storagePermissionCallback");
            }
        }
    }

    @ActivityCallback
    private void manageStoragePermissionResult(PluginCall call, ActivityResult result) {
        boolean granted = Environment.isExternalStorageManager();
        call.resolve(permManager.createPermissionResult(granted, 
            granted ? "MANAGE_EXTERNAL_STORAGE permission granted" : "MANAGE_EXTERNAL_STORAGE permission denied"));
    }

    @PluginMethod
    public void checkPermissions(PluginCall call) {
        permManager.checkPermissions(call);
    }

    @PermissionCallback
    private void storagePermissionCallback(PluginCall call) {
        boolean granted = permManager.hasStoragePermissions();
        call.resolve(permManager.createPermissionResult(granted,
            granted ? "Storage permissions granted" : "Storage permissions denied"));
    }

    // ==================== 目录操作 ====================

    @PluginMethod
    public void listDirectory(PluginCall call) {
        String path = call.getString("path");
        Boolean showHidden = call.getBoolean("showHidden", false);
        String sortBy = call.getString("sortBy", "name");
        String sortOrder = call.getString("sortOrder", "asc");

        if (path == null) {
            call.reject("Path is required");
            return;
        }

        try {
            JSObject result = dirOps.listDirectory(path, showHidden, sortBy, sortOrder);
            call.resolve(result);
        } catch (Exception e) {
            call.reject("Failed to list directory: " + e.getMessage());
        }
    }

    @PluginMethod
    public void createDirectory(PluginCall call) {
        String path = call.getString("path");
        Boolean recursive = call.getBoolean("recursive", false);

        if (path == null) {
            call.reject("Path is required");
            return;
        }

        try {
            dirOps.createDirectory(path, recursive);
            call.resolve();
        } catch (Exception e) {
            call.reject("Failed to create directory: " + e.getMessage());
        }
    }

    @PluginMethod
    public void deleteDirectory(PluginCall call) {
        String path = call.getString("path");

        if (path == null) {
            call.reject("Path is required");
            return;
        }

        try {
            dirOps.deleteDirectory(path);
            call.resolve();
        } catch (Exception e) {
            call.reject("Failed to delete directory: " + e.getMessage());
        }
    }

    // ==================== 文件操作 ====================

    @PluginMethod
    public void createFile(PluginCall call) {
        String path = call.getString("path");
        String content = call.getString("content", "");
        String encoding = call.getString("encoding", "utf8");

        if (path == null) {
            call.reject("Path is required");
            return;
        }

        try {
            fileOps.createFile(path, content, encoding);
            call.resolve();
        } catch (Exception e) {
            call.reject("Failed to create file: " + e.getMessage());
        }
    }

    @PluginMethod
    public void readFile(PluginCall call) {
        String path = call.getString("path");
        String encoding = call.getString("encoding", "utf8");

        if (path == null) {
            call.reject("Path is required");
            return;
        }

        try {
            JSObject result = fileOps.readFile(path, encoding);
            call.resolve(result);
        } catch (Exception e) {
            call.reject("Failed to read file: " + e.getMessage());
        }
    }

    @PluginMethod
    public void writeFile(PluginCall call) {
        String path = call.getString("path");
        String content = call.getString("content");
        String encoding = call.getString("encoding", "utf8");
        Boolean append = call.getBoolean("append", false);

        if (path == null || content == null) {
            call.reject("Path and content are required");
            return;
        }

        try {
            fileOps.writeFile(path, content, encoding, append);
            call.resolve();
        } catch (Exception e) {
            call.reject("Failed to write file: " + e.getMessage());
        }
    }

    @PluginMethod
    public void deleteFile(PluginCall call) {
        String path = call.getString("path");

        if (path == null) {
            call.reject("Path is required");
            return;
        }

        try {
            fileOps.deleteFile(path);
            call.resolve();
        } catch (Exception e) {
            call.reject("Failed to delete file: " + e.getMessage());
        }
    }

    @PluginMethod
    public void renameFile(PluginCall call) {
        String path = call.getString("path");
        String newName = call.getString("newName");

        if (path == null || newName == null) {
            call.reject("Path and newName are required");
            return;
        }

        try {
            fileOps.renameFile(path, newName);
            call.resolve();
        } catch (Exception e) {
            call.reject("Failed to rename file: " + e.getMessage());
        }
    }

    @PluginMethod
    public void moveFile(PluginCall call) {
        String sourcePath = call.getString("sourcePath");
        String destinationPath = call.getString("destinationPath");

        if (sourcePath == null || destinationPath == null) {
            call.reject("sourcePath and destinationPath are required");
            return;
        }

        try {
            fileOps.moveFile(sourcePath, destinationPath);
            call.resolve();
        } catch (Exception e) {
            call.reject("Failed to move file: " + e.getMessage());
        }
    }

    @PluginMethod
    public void copyFile(PluginCall call) {
        String sourcePath = call.getString("sourcePath");
        String destinationPath = call.getString("destinationPath");
        Boolean overwrite = call.getBoolean("overwrite", false);

        if (sourcePath == null || destinationPath == null) {
            call.reject("sourcePath and destinationPath are required");
            return;
        }

        try {
            fileOps.copyFile(sourcePath, destinationPath, overwrite);
            call.resolve();
        } catch (Exception e) {
            call.reject("Failed to copy file: " + e.getMessage());
        }
    }

    @PluginMethod
    public void getFileInfo(PluginCall call) {
        String path = call.getString("path");

        if (path == null) {
            call.reject("Path is required");
            return;
        }

        try {
            JSObject result = fileOps.getFileInfo(path);
            call.resolve(result);
        } catch (Exception e) {
            call.reject("Failed to get file info: " + e.getMessage());
        }
    }

    @PluginMethod
    public void exists(PluginCall call) {
        String path = call.getString("path");

        if (path == null) {
            call.reject("Path is required");
            return;
        }

        JSObject result = new JSObject();
        result.put("exists", fileOps.exists(path));
        call.resolve(result);
    }

    // ==================== 搜索 ====================

    @PluginMethod
    public void searchFiles(PluginCall call) {
        String directory = call.getString("directory");
        String query = call.getString("query");
        String searchType = call.getString("searchType", "name");
        JSArray fileTypesArray = call.getArray("fileTypes");
        Integer maxResults = call.getInt("maxResults", 100);
        Boolean recursive = call.getBoolean("recursive", true);

        if (directory == null || query == null) {
            call.reject("Directory and query are required");
            return;
        }

        try {
            String[] fileTypes = null;
            if (fileTypesArray != null && fileTypesArray.length() > 0) {
                fileTypes = new String[fileTypesArray.length()];
                for (int i = 0; i < fileTypesArray.length(); i++) {
                    fileTypes[i] = fileTypesArray.getString(i);
                }
            }

            JSObject result = fileSearcher.searchFiles(directory, query, searchType, 
                                                        fileTypes, maxResults, recursive);
            call.resolve(result);
        } catch (Exception e) {
            call.reject("Failed to search files: " + e.getMessage());
        }
    }

    // ==================== 系统文件选择器 ====================

    @PluginMethod
    public void openSystemFilePicker(PluginCall call) {
        String type = call.getString("type", "file");
        Boolean multiple = call.getBoolean("multiple", false);
        JSArray acceptArray = call.getArray("accept");
        String title = call.getString("title", "选择文件");

        filePicker.openFilePicker(call, type, multiple, acceptArray, title);
    }

    @ActivityCallback
    private void filePickerResult(PluginCall call, ActivityResult result) {
        JSObject ret = new JSObject();
        JSArray files = new JSArray();
        JSArray directories = new JSArray();

        if (result.getResultCode() == getActivity().RESULT_OK && result.getData() != null) {
            Intent data = result.getData();
            String requestType = call.getString("type", "file");

            // 处理多选
            if (data.getClipData() != null) {
                int count = data.getClipData().getItemCount();
                for (int i = 0; i < count; i++) {
                    Uri uri = data.getClipData().getItemAt(i).getUri();
                    JSObject fileInfo = filePicker.getFileInfoFromUri(uri);
                    if (fileInfo != null) {
                        if ("directory".equals(fileInfo.getString("type"))) {
                            directories.put(fileInfo);
                        } else {
                            files.put(fileInfo);
                        }
                    }
                }
            } else if (data.getData() != null) {
                // 处理单选
                Uri uri = data.getData();
                JSObject fileInfo = filePicker.getFileInfoFromUri(uri);
                if (fileInfo != null) {
                    if ("directory".equals(requestType) || "directory".equals(fileInfo.getString("type"))) {
                        directories.put(fileInfo);
                    } else {
                        files.put(fileInfo);
                    }
                }
            }

            ret.put("files", files);
            ret.put("directories", directories);
            ret.put("cancelled", false);
        } else {
            ret.put("files", files);
            ret.put("directories", directories);
            ret.put("cancelled", true);
        }

        call.resolve(ret);
    }

    @PluginMethod
    public void openSystemFileManager(PluginCall call) {
        String path = call.getString("path", "/storage/emulated/0");

        try {
            Intent intent = new Intent(Intent.ACTION_VIEW);
            Uri uri = Uri.parse(path);
            intent.setDataAndType(uri, "resource/folder");
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);

            if (intent.resolveActivity(getContext().getPackageManager()) != null) {
                getActivity().startActivity(intent);
            } else {
                Intent fallbackIntent = new Intent(Intent.ACTION_GET_CONTENT);
                fallbackIntent.setType("*/*");
                fallbackIntent.addCategory(Intent.CATEGORY_OPENABLE);
                fallbackIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                getActivity().startActivity(Intent.createChooser(fallbackIntent, "打开文件管理器"));
            }
            call.resolve();
        } catch (Exception e) {
            call.reject("Failed to open system file manager: " + e.getMessage());
        }
    }

    @PluginMethod
    public void openFileWithSystemApp(PluginCall call) {
        String filePath = call.getString("path");
        String mimeType = call.getString("mimeType");

        if (filePath == null) {
            call.reject("File path is required");
            return;
        }

        try {
            File file = new File(filePath);
            if (!file.exists()) {
                call.reject("File does not exist: " + filePath);
                return;
            }

            Uri uri;
            try {
                uri = FileProvider.getUriForFile(getContext(),
                        getContext().getPackageName() + ".fileprovider", file);
            } catch (Exception e) {
                uri = Uri.fromFile(file);
            }

            Intent intent = new Intent(Intent.ACTION_VIEW);

            if (mimeType == null || mimeType.isEmpty()) {
                mimeType = FileUtils.getMimeType(filePath);
            }

            intent.setDataAndType(uri, mimeType);
            intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);

            getActivity().startActivity(Intent.createChooser(intent, "打开文件"));
            call.resolve();
        } catch (Exception e) {
            call.reject("Failed to open file: " + e.getMessage());
        }
    }

    // ==================== AI 编辑操作 ====================

    @PluginMethod
    public void readFileRange(PluginCall call) {
        String path = call.getString("path");
        Integer startLine = call.getInt("startLine");
        Integer endLine = call.getInt("endLine");

        if (path == null || startLine == null || endLine == null) {
            call.reject("path, startLine and endLine are required");
            return;
        }

        try {
            JSObject result = aiEditOps.readFileRange(path, startLine, endLine);
            call.resolve(result);
        } catch (Exception e) {
            call.reject("Failed to read file range: " + e.getMessage());
        }
    }

    @PluginMethod
    public void insertContent(PluginCall call) {
        String path = call.getString("path");
        Integer line = call.getInt("line");
        String content = call.getString("content");

        if (path == null || line == null || content == null) {
            call.reject("path, line and content are required");
            return;
        }

        try {
            aiEditOps.insertContent(path, line, content);
            call.resolve();
        } catch (Exception e) {
            call.reject("Failed to insert content: " + e.getMessage());
        }
    }

    @PluginMethod
    public void replaceInFile(PluginCall call) {
        String path = call.getString("path");
        String search = call.getString("search");
        String replace = call.getString("replace");
        Boolean isRegex = call.getBoolean("isRegex", false);
        Boolean replaceAll = call.getBoolean("replaceAll", true);
        Boolean caseSensitive = call.getBoolean("caseSensitive", true);

        if (path == null || search == null || replace == null) {
            call.reject("path, search and replace are required");
            return;
        }

        try {
            JSObject result = aiEditOps.replaceInFile(path, search, replace, 
                                                       isRegex, replaceAll, caseSensitive);
            call.resolve(result);
        } catch (Exception e) {
            call.reject("Failed to replace in file: " + e.getMessage());
        }
    }

    @PluginMethod
    public void applyDiff(PluginCall call) {
        String path = call.getString("path");
        String diff = call.getString("diff");
        Boolean createBackup = call.getBoolean("createBackup", false);

        if (path == null || diff == null) {
            call.reject("path and diff are required");
            return;
        }

        try {
            JSObject result = aiEditOps.applyDiff(path, diff, createBackup);
            call.resolve(result);
        } catch (Exception e) {
            call.reject("Failed to apply diff: " + e.getMessage());
        }
    }

    @PluginMethod
    public void getFileHash(PluginCall call) {
        String path = call.getString("path");
        String algorithm = call.getString("algorithm", "md5");

        if (path == null) {
            call.reject("path is required");
            return;
        }

        try {
            JSObject result = aiEditOps.getFileHash(path, algorithm);
            call.resolve(result);
        } catch (Exception e) {
            call.reject("Failed to get file hash: " + e.getMessage());
        }
    }

    @PluginMethod
    public void getLineCount(PluginCall call) {
        String path = call.getString("path");

        if (path == null) {
            call.reject("path is required");
            return;
        }

        try {
            JSObject result = aiEditOps.getLineCount(path);
            call.resolve(result);
        } catch (Exception e) {
            call.reject("Failed to get line count: " + e.getMessage());
        }
    }

    // ==================== 工具方法 ====================

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");
        JSObject ret = new JSObject();
        ret.put("value", value);
        call.resolve(ret);
    }
}
